File: frontend/src/models/viewmodels/place.vm.ts
import { PaginationData } from "../app.interface";
import { CategoryVM } from "./category.vm";
import { CityVM } from "./city.vm";

export interface PlaceVM {
  id: number;
  name: string;
  shortDescription: string;
  cover: string;
  dateCreated: string;
  city: string;
  ownerId: number;
  ownerAvatar: string;
  ownerName: string;
}

export interface PlaceDetailVM {
  id: number;
  name: string;
  shortDescription: string;
  description: string;
  images: string[];
  address: string;
  viewCount: number;

  // Relationship
  city: CityVM;
  ownerId: number;
  ownerAvatar: string;
  ownerName: string;
  dateCreated: string;
  categories: CategoryVM[];
}

export interface PagedPlacesVM {
  places: PlaceVM[];
  pagination: PaginationData;
}

File: backend/Common/ViewModels/PlaceVMs/PlaceVM.cs
namespace Common.ViewModels.PlaceVMs;
public class PlaceVM
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string ShortDescription { get; set; }
    public string Cover { get; set; }
    public string DateCreated { get; set; }
    public string City { get; set; }
    public int OwnerId { get; set; }
    public string OwnerAvatar { get; set; }
    public string OwnerName { get; set; }
}
